### ISSUE TITLE
Design better boolean arguments

### ISSUE THREAD

**Issue by matejak** [matejak/argbash#14]
Currently, the `ARG_OPTIONAL_BOOL` does not behave optimally (see #2 ).
It seems to be a good idea to have more macros for switch-on, switch-off and both (as the current `ARG_OPTIONAL_BOOL` somehow attempts to).
Currently, `ARG_OPTIONAL_BOOL` assumes that one wants to switch something on (using long and short option) and autogenerating long option to switch something off. This falls on its head when one wants to switch something off.
Proposed behavior:
* `ARG_OPTIONAL_BOOL` will remain, it will autodetect whether it is in a switch-on or switch-off mode by examining the default (or the option whether it begins with`no-...`). It will use the provided long and short options for the detected mode, but it will also generate a long option for the opposite mode.
New macros would be introduced:
* `ARG_OPTIONAL_SWITCH_ON` will be introduced. It will accept long and short option (and no default since `off` will be assumed as default).
* `ARG_OPTIONAL_SWITCH_OFF` will be introduced. It will accept long and short option (and no default since `on` will be assumed).
Everybody's comments will be highly appreciated!

**Comment by edannenberg** [comment]
Sorry for the delay, my 2 cents:

I think `ARG_OPTIONAL_BOOL` should be renamed to `ARG_OPTIONAL_BOOL_NULLABLE`, it would be more precise. A real bool should only have 2 possible states, probably what user's are after 99% of the time. As I said in the other issue I have a hard time imagining a use case for a nullable bool. It should be rather rare.

The proposed `ARG_OPTIONAL_SWITCH_ON` should become `ARG_OPTIONAL_BOOL`, should also prevent most BC issues. I'm not sure `ARG_OPTIONAL_SWITCH_OFF` is needed, setting a default value for  `ARG_OPTIONAL_BOOL` should take care of everything else.

**Comment by matejak** [comment]
Good point, (any) code should read like a prose, so letting `ARG_OPTIONAL_SWITCH_OFF(video, v)` to create the `--no-video` option while saving the result to `_arg_video` should do the trick. There could be an option to set the prefix (default would be `no-`, people may be interested in `disable-` for `--disable-video` etc.).

**Comment by 0x5c** [comment]
Hello!
Is this still in the plans? The milestone is 2.7.0, but 2.8.0 got released a month ago.
It's what would make argbash perfect for me.

**Comment by matejak** [comment]
Hello, yes, it is deffinitelly planned. Stay tuned for updates!

**Comment by cconnert** [comment]
@matejak: Hi, will this issue be resolved? Seems like you got pretty far on your draft. Thanks

**Comment by ShamrockLee** [comment]
It would also be very helpful to have a way to integrate with [arithmetic condition](https://tiswww.case.edu/php/chet/bash/bashref.html#Conditional-Constructs) `(( expression ))` in modern Bash, which returns successfully (status 0) when `expression` evaluates to nonzero, or else returns failure (status 1).

Something like `ARG_OPTIONAL_BOOL_NUM`, `ARG_SWITCH_YES_NUM`, `ARG_SWITCH_NO_NUM` would be great.

Here's the behavior of Bash arithmetic conditional construct:
```console
$ if (( 1 )); then echo yes; else echo no; fi
yes
$ if (( 0 )); then echo yes; else echo no; fi
no
$ if (( "" )); then echo yes; else echo no; fi
no
$ if (( )); then echo yes; else echo no; fi
no
$ unset foo; if (( foo )); then echo yes; else echo no; fi
no
$ foo=2; if (( foo )); then echo yes; else echo no; fi
yes
$ foo=2; if (( $foo )); then echo yes; else echo no; fi
yes
$ foo=2; if (( "$foo" )); then echo yes; else echo no; fi
yes
```
